server/service/src/sync/central_data_synchroniser_v6.rs

```diff
@@ -0,0 +1,97 @@
+use crate::sync::{
+    api_v6::{SyncBatchV6, SyncRecordV6},
+    sync_status::logger::SyncStepProgress,
+};
+
+use super::{
+    api::ParsingSyncRecordError,
+    api_v6::{SyncApiErrorV6, SyncApiV6},
+    sync_status::logger::{SyncLogger, SyncLoggerError},
+};
+
+use repository::{
+    CursorController, KeyValueType, RepositoryError, StorageConnection, SyncBufferRow,
+    SyncBufferRowRepository,
+};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub(crate) enum CentralPullErrorV6 {
+    #[error(transparent)]
+    SyncApiError(#[from] SyncApiErrorV6),
+    #[error("Failed to save sync buffer or cursor")]
+    SaveSyncBufferOrCursorsError(#[from] RepositoryError),
+    #[error(transparent)]
+    ParsingRecordError(#[from] ParsingSyncRecordError),
+    #[error(transparent)]
+    SyncLoggerError(#[from] SyncLoggerError),
+}
+
+#[derive(Error, Debug)]
+#[error("Failed to serialise V6 remote record into sync buffer row, record: '{record:?}'")]
+pub(crate) struct SerialisingToSyncBuffer {
+    source: serde_json::Error,
+    record: serde_json::Value,
+}
+
+pub(crate) struct CentralDataSynchroniserV6 {
+    pub(crate) sync_api_v6: SyncApiV6,
+}
+
+impl CentralDataSynchroniserV6 {
+    pub(crate) async fn pull<'a>(
+        &self,
+        connection: &StorageConnection,
+        batch_size: u32,
+        logger: &mut SyncLogger<'a>,
+    ) -> Result<(), CentralPullErrorV6> {
+        let cursor_controller = CursorController::new(KeyValueType::SyncPullCursorV6);
+        // TODO protection from infinite loop
```

Comment:
Is there an issue for this TODO?

Reply:
Nope, this comment has been there already, i just added to [original sync carry over](https://github.com/msupply-foundation/open-msupply/issues/608), thanks